src/woo_publications/publications/tests/test_document_api_endpoints.py

```diff
@@ -262,9 +252,18 @@ def test_list_documents_filter_owner(self):
                 headers=AUDIT_HEADERS,
             )
 
-            self.assertEqual(response.status_code, status.HTTP_200_OK)
+            self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
             data = response.json()
-            self.assertEqual(data["count"], 0)
+            self.assertEqual(
+                data["eigenaar"],
+                [
+                    _(
+                        "Select a valid choice. That choice is not one "
+                        "of the available choices."
+                    )
+                ],
+            )
```

Comment:
can't we keep the old behaviour? Dealing with an empty result list on the client side is no problem, but handling HTTP 400 errors now requires changes on the API consumer, which is a breaking change.

I think we can keep defining a text filter, but instead give the lookup to `eigenaar__identifier__exact` instead of using a relational field, no? That should also resolve our API schema generation problem.

Reply:
Additionally, this validation error is a bit of an enumeration attack vulnerability - you can exactly determine which internal user identifiers may exist, while with an empty result list you can't infer whether it exists or not, or if there are just no publications/documents owned by that person.